### ISSUE TITLE
Touchbar buttons lacks state awareness

### ISSUE THREAD

**Issue by jbergstroem** [salomvary/soundcleod#154]
The touchbar icons seems to lack the state of the player; eg play/pause (is track playing) or if a track is liked.

**Comment by salomvary** [comment]
Play/pause state works for me. If it still seems broken for you, please provide more details.

Like state never worked (it definitely should). @rmaes4 did some work to implement like state management on the touch bar but we never got to finish it.

**Comment by rmaes4** [comment]
Sorry I’ve been busy. Maybe I could take a look this weekend

**Comment by jbergstroem** [comment]
> @salomvary said:
> Play/pause state works for me. If it still seems broken for you, please provide more details.

Yep, can confirm. It's just state (are we playing or are we paused) that doesn't.

**Comment by greatestape** [comment]
I'm having this issue, or some cousin to it.

* When the touchbar button is controlling Soundcleod, it doesn't change state when I press it, but it does toggle play/pause in Soundcleod
* When iTunes is playing music, play/pause still toggles Soundcleod and doesn't affect iTunes
* I think I had this issue with the regular play/pause button on my older Macbook.

Let me know if you would like any logs or videos or whatever that would help diagnose this.

**Comment by jbergstroem** [comment]
I realize now that I didn't express myself very well. I was specifically referring to the `MediaPlayPause`,`MediaPlayNext`,`MediaPlayPrevious` global shortcuts and not the buttons provided by Soundcleod.

**Comment by apeeters** [comment]
Consider providing a similar behaviour as Google Play Music Desktop Player, their component is available as a library: https://github.com/MarshallOfSound/electron-media-service

**Comment by salomvary** [comment]
Thanks for the tip @apeeters, looks like using this library would be a good improvement (also solves ##179).

**Comment by jbergstroem** [comment]
Perhaps the native mediakey support in chrome 73 can find its way back to electron as well: https://developers.google.com/web/updates/2019/02/chrome-73-media-updates

..too bad electron 5 (release in 3 weeks) is targeting chrome 72.
